### ISSUE TITLE
[ISSUE]: 

### ISSUE THREAD

**Issue by TheRatzzz** [w-okada/voice-changer#1297]
### Voice Changer Version

MMVCServerSIO_win_onnxdirectML-cuda_v.1.5.3.17b

### Operational System

Windows 10

### GPU

Nvidia GeForce 1650

### Read carefully and check the options

- [X] I've tried to Clear Settings
- [ ] Sample/Default Models are working
- [X] I've tried to change the Chunk Size
- [X] GUI was successfully launched
- [X] I've read the [tutorial](https://github.com/w-okada/voice-changer/blob/master/tutorials/tutorial_rvc_en_latest.md)
- [ ] I've tried to extract to another folder (or re-extract) the .zip file

### Model Type

RVC

### Issue Description

When i start a game the audio wont pick up, or its gets very laggy and cuts out alot.

### Logs on console

C:\Users\carro\OneDrive\Desktop\AI>MMVCServerSIO.exe     -p 18888     --https false     --content_vec_500 pretrain/checkpoint_best_legacy_500.pt     --content_vec_500_onnx pretrain/content_vec_500.onnx     --content_vec_500_onnx_on true     --hubert_base pretrain/hubert_base.pt     --hubert_base_jp pretrain/rinna_hubert_base_jp.pt     --hubert_soft pretrain/hubert/hubert-soft-0d54a1f4.pt     --nsf_hifigan pretrain/nsf_hifigan/model     --crepe_onnx_full pretrain/crepe_onnx_full.onnx     --crepe_onnx_tiny pretrain/crepe_onnx_tiny.onnx     --rmvpe pretrain/rmvpe.pt     --model_dir model_dir     --samples samples.json
    Booting PHASE :__main__
    PYTHON:3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
    Activating the Voice Changer.
[Voice Changer] download sample catalog. samples_0004_t.json
[Voice Changer] download sample catalog. samples_0004_o.json
[Voice Changer] download sample catalog. samples_0004_d.json
[Voice Changer] model_dir is already exists. skip download samples.
    Internal_Port:18888
    protocol: HTTP
    -- ---- --
    Please open the following URL in your browser.
    http://<IP>:<PORT>/
    In many cases, it will launch when you access any of the following URLs.
    http://127.0.0.1:18888/

[VCClient] Access http://127.0.0.1:18888/
[VCClient] wait web server...0 http://127.0.0.1:18888/
    Booting PHASE :__main__
    Booting PHASE :MMVCServerSIO
[Voice Changer] VoiceChangerManager initializing...
[Voice Changer] model slot is changed -1 -> 4
................RVC
[Voice Changer] [RVCr2] Creating instance
VoiceChangerV2 Initialized (GPU_NUM(cuda):1, mps_enabled:False, onnx_device:CPU-DML)
[Voice Changer][RVC]: update_settings gpu:0
[Voice Changer][RVCr2] Initializing...
gin_channels: 256 self.spk_embed_dim: 109
[Voice Changer] generate new embedder. (no embedder)
2024-08-03 03:14:30.4451105 [W:onnxruntime:, session_state.cc:1169 onnxruntime::VerifyEachNodeIsAssignedToAnEp] Some nodes were not assigned to the preferred execution providers which may or may not have an negative impact on performance. e.g. ORT explicitly assigns shape related ops to CPU to improve perf.
2024-08-03 03:14:30.4512152 [W:onnxruntime:, session_state.cc:1171 onnxruntime::VerifyEachNodeIsAssignedToAnEp] Rerunning with verbose output on a non-minimal build will show node assignments.
[Voice Changer] Loading index...
[Voice Changer] Index file is not found
GENERATE INFERENCER<voice_changer.RVC.inferencer.RVCInferencerv2.RVCInferencerv2 object at 0x000001F41A483F10>
GENERATE EMBEDDER<voice_changer.RVC.embedder.OnnxContentvec.OnnxContentvec object at 0x000001F423677DF0>
GENERATE PITCH EXTRACTOR<voice_changer.RVC.pitchExtractor.RMVPEOnnxPitchExtractor.RMVPEOnnxPitchExtractor object at 0x000001F423677E20>
[Voice Changer] [RVC] Initializing... done
[Voice Changer][RVC]: update_settings serverReadChunkSize:24
[Voice Changer][RVC]: update_settings extraConvertSize:4096
[Voice Changer][RVC]: update_settings silentThreshold:0
[Voice Changer][RVC]: update_settings f0Detector:rmvpe_onnx
[Voice Changer][RVC]: update_settings modelSlotIndex:1722599420004
[Voice Changer] VoiceChangerManager initializing... done.
[Voice Changer] MMVC_Rest initializing...
[Voice Changer] MMVC_Rest initializing... done.
[Voice Changer] MMVC_SocketIOApp initializing...
[Voice Changer] MMVC_SocketIOApp initializing... done.
[VCClient] wait web server... done 200
[2024-08-03 03:14:35] connet sid : tNTPu5ztAAqj0NEwAAAB
[2024-08-03 03:14:35] connet sid : t9V_jkemVG-FJfWiAAAD

**Comment by Kuuko-fokkusugaru** [comment]
You need a better GPU to be able to run RVC and a game at the same time.

**Comment by Kuuko-fokkusugaru** [comment]
You should also download the latest version instead of .17b. Latest one ends in 18a instead. Or you could try v2, but performance may be worse.
Also, download the nvidia one instead of the direct ml version.

It's still unlikely to be able to play games and use RVC at the same time with that GPU because that's not even a gaming GPU. But give it a try to the above just in case.

**Comment by TheRatzzz** [comment]
I tried both and it just gives up when i start games, but work perfectly when i have nothing else open.

**Comment by YourLvurr** [comment]
> You need a better GPU to be able to run RVC and a game at the same time.

This is not true, I have an RTX 4060 and I have the same issue, the audio cuts out as soon as I enter any game

**Comment by Kuuko-fokkusugaru** [comment]
> > You need a better GPU to be able to run RVC and a game at the same time.
> 
> This is not true, I have an RTX 4060 and I have the same issue, the audio cuts out as soon as I enter any game

It's actually 200% true. The person that opened the issue owns an nvidia 1650 which isn't even meant for gaming. And your issue is actually related as well.

RVC uses a certain % of the GPU while you speak which you can monitor easily with the task manager open. This % of GPU is relative to the performance of each GPU. So a lower end GPU will have a higher GPU usage than a higher end one. For example, we have users here that reported a 40-90% of GPU usage with a 4070 and you are using a 4060 which is way inferior. If RVC use, let's say, a 40% of your GPU and a game uses a 80%, you are missing a 20% of performance needed to run both smoothly at the same time. Whether you can se RVC with a game will depend a lot on which game are you trying to play and how much GPU it needs. Keep in mind that RVC main purpose is to be used with a secondary computer used as a server and not by the same computer where you are running your games. That said, if you have a strong enough GPU, you can run both depending on the game and how demanding it is. I use a 3080Ti which is strong enough to pull RVC and most games but I can't either expect those games to run at more than 60fps if I don't want RVC to get a huge delay or stutter.

If you want a GPU capable of running games and RVC together, you need to look into the highest end ones. Basically, as of right now, 3080Ti, 3090, 4080, 4080 Super, 4090. You need to look for those GPUs with the higher CUDA cores count. A 4060 is not meant for computational tasks after all.
